src/libstd/ffi/os_str.rs

```diff
@@ -512,6 +528,11 @@ impl OsStr {
         unsafe { &*(inner as *const Slice as *const OsStr) }
     }
 
+    #[inline]
+    fn from_inner_mut(inner: &mut Slice) -> &mut OsStr {
+        unsafe { &mut *(inner as *mut Slice as *mut OsStr) }
```

Comment:
Can you add some comments re. the unsafety here and in the cases below (other files in diff)?

Reply:
I can. The unsafety here is essentially the same as the unsafety on `from_inner` and related. Should I add comments for both `from_inner` and `from_inner_mut` ?

Reply:
I added some comments to both `from_inner` and `from_inner_mut` as well as related methods.

Reply:
Thanks!